src/components/SessionVerification.tsx

```diff
@@ -0,0 +1,22 @@
+import useSWR from 'swr'
+
+export const LOADING_STATE = 'loading'
+const fetcher = (url: string): Promise<unknown> =>
+  // eslint-disable-next-line unicorn/prevent-abbreviations
+  fetch(url).then((res) => res.json())
+export const useSessionToken = (baseUrl: string): string => {
+  const { data, error } = useSWR(baseUrl, fetcher)
+
+  if (error) {
```

Comment:
return user directly no need for custom extract logic

Reply:
not sure I understand...if there's an error returned, don't we want to redirect the user to home page?

Reply:
coz if we remove this, we will not be able to distinguish between loading state and error